src/Entity/EntityId.php

```diff
@@ -179,4 +203,18 @@ public function equals( $target ) {
 			&& $target->serialization === $this->serialization;
 	}
 
+	/**
+	 * Returns a pair (repository name, local part of ID) from the given ID serialization.
+	 * Note: this does not perform any validation of the given input. Calling code should take
+	 * care of this!
+	 *
+	 * @param string $serialization
+	 * @return string[] Array of form [ string $repositoryName, string $localPart ]
+	 */
+	protected static function extractRepositoryNameAndLocalPart( $serialization ) {
+		list( $repoName, $prefixRemainder, $localId ) = self::extractSerializationParts( $serialization );
+		$localPart = self::joinSerialization( [ '', $prefixRemainder, $localId ] );
```

Comment:
The fact that the two last elements are never used independent from each other makes this look like it should be optimized even further: Just do `explode( ':', $serialization, 2 )` and return the second element. Done.

Reply:
IIRC, we considered this when we first designed this class, and we found that we sometimes need only the first element, and sometimes only the last. But I don't remember why, exactly, but there was *some* reason.

Reply:
The reason why splitSerialization does return always three parts was indeed that sometimes caller might want first element and second+third together, sometimes it might though only want third element separately.

In this case though, what @thiemowmde suggest would be fine. Only possible concern (I am not saying it is) is then the detail about splitting by colon is in two places instead of just one. Having the separator details encapsulated in a single method used to be considered important if I recall correctly.